File: explorer/darwin/ios/lynx_test_bench/TestBenchEnv.m
// Copyright 2025 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#import "TestBenchEnv.h"

@interface TestBenchEnv ()

@end

@implementation TestBenchEnv

+ (instancetype)sharedInstance {
  static TestBenchEnv *sharedInstance = nil;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^{
    sharedInstance = [[self alloc] init];
  });
  return sharedInstance;
}

- (instancetype)init {
  if (self = [super init]) {
    self.testBenchUrlPrefix = @"file://testbench?";
    self.testBenchUrlSchema = @"file";
    self.testBenchUrlHost = @"testbench";
  }
  return self;
}
@end

File: explorer/android/lynx_test_bench/src/main/java/com/lynx/testbench/TestBenchEnv.java
// Copyright 2024 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

package com.lynx.testbench;

public class TestBenchEnv {
  private static TestBenchEnv instance;
  public String mUriKey = "uri";
  public String mTestBenchUrlPrefix = "file://testbench?";

  public static synchronized TestBenchEnv getInstance() {
    if (instance == null) {
      instance = new TestBenchEnv();
    }
    return instance;
  }
}
